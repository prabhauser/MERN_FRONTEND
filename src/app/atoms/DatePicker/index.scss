@import '../../../utilities/theme/_theme.scss';

.base-copy {
    box-sizing: border-box !important;
    width: 400px !important;
    border: 1px solid $s100Bg !important;
    border-radius: 10px !important;
    background-color: $white !important;
    height: 44px;
}

.DatePickaer {
    box-sizing: border-box !important;
    height: 44px !important;
    width: 400px !important;
    border: 1px solid $s100Bg !important;
    border-radius: 10px !important;
    background-color: $white !important;
}

.base-copy .MuiOutlinedInput-root {
    border-radius: 10px !important;
}
.lable {
    height: 17px;
    opacity: 0.5;
    color: #2c333a;
    font-family: Poppins;
    font-size: 12px;
    font-weight: 500;
    letter-spacing: 0;
    margin-bottom: 12px;
}
