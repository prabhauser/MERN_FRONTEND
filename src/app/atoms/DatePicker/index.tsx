import TextField from '@mui/material/TextField';
// import AdapterDateFns from '@mui/lab/AdapterDateFns';
// import LocalizationProvider from '@mui/lab/LocalizationProvider';
// import DatePicker from '@mui/lab/DatePicker';
import { makeStyles } from '@material-ui/core/styles';
import './index.scss';
// import { ClassNames } from '@emotion/react';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { DatePicker } from '@mui/x-date-pickers/DatePicker';

const useStyles = makeStyles(() => ({
    root: {
        '& .MuiOutlinedInput-root': {
            borderRadius: '10px',
            '& .MuiInputBase-input': {
                padding: '18px'
            },
            '& legend': {
                width: '0%'
            }
        }
    }
}));

type ComponentProps = {
    name?: string;
    value: any;
    handleChange?: any;
    disabled?: boolean;
    showLabel?: boolean;
    busFormClass?: boolean;
    customLabel?: boolean;
    customLabelVal?: any;
};

export default function BasicDatePicker(props: ComponentProps) {
    const {
        name = '',
        value,
        handleChange,
        disabled,
        showLabel,
        busFormClass,
        customLabel = false,
        customLabelVal
    } = props;
    const classes = useStyles();
    const date = new Date();
    return (
        <LocalizationProvider dateAdapter={AdapterDateFns}>
            {showLabel ? <span className="lable">Filter By Date</span> : null}
            {customLabel ? customLabelVal : null}
            <DatePicker
                value={value}
                onChange={handleChange}
                disabled={disabled}
                disableFuture
                inputFormat="dd/MM/yyyy"
                renderInput={(params) => (
                    <TextField
                        name={name}
                        disabled={disabled}
                        {...params}
                        sx={{ width: '100%' }}
                        className={!busFormClass ? classes.root : ''}
                    />
                )}
            />
        </LocalizationProvider>
    );
}
