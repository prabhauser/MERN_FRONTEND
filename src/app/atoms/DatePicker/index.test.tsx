import { shallow } from 'enzyme';
import DatePicker from './index';

describe('<DatePicker />', () => {
    it('should take DatePicker snapshot', () => {
        const component = shallow(<DatePicker />);
        component.update();
        expect(component).toMatchSnapshot();
    });
});
