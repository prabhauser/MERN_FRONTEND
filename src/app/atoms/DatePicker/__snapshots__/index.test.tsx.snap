// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> should take DatePicker snapshot 1`] = `
<LocalizationProvider
  dateAdapter={[Function]}
>
  <ForwardRef(DatePicker)
    disableFuture={true}
    inputFormat="dd/MM/yyyy"
    renderInput={[Function]}
  />
</LocalizationProvider>
`;
